src/kayenta/domain/ICanaryConfig.ts

```diff
@@ -22,6 +22,7 @@ export interface ICanaryMetricConfig {
   groups: string[];
   analysisConfigurations: {
     [key: string]: any;
+    effectSize?: ICanaryMetricEffectSizeConfig;
```

Comment:
I think `analysisConfigurations` was left untyped because we never addressed the problem of having separate canary judges with different analysis configuration properties. Leaving `effectSize` as nullable seems fine for now.

Reply:
My current, loosely held opinion: even if the various judges have wildly different judge specific fields, it's likely still worth a best effort attempt to type both the shared and judge specific fields as they relate to the UI (i.e. we may not end up reflecting the full range of shapes that the backend can handle). I have no interest in working against the dynamic shape of `analysisConfigurations` in general, but it's helpful to get autocomplete and identify when a well defined field is being used incorrectly. If we did this for more properties I'm not sure what it would look like — lots of nullable fields, or shared and judge specific interfaces pulled together with `Partial`, perhaps? 🤷‍♂️